p11-kit/list-objects.c

```diff
@@ -192,16 +189,16 @@ get_object_uri (P11KitIter *iter,
 		goto cleanup;
 	}
 
-	if (label != NULL) {
-	        ret = p11_kit_uri_set_attribute (uri, &attr_label);
+	if (attr_label != NULL) {
```

Comment:
These will always be true in this use case. The idea was to not set attributes which have `pValue == NULL`. I would suggest adding a check for `attr->pValue != NULL` as well. However I dont know if setting a `NULL` value attr matters for URI.

Reply:
I think in this case `pValue` is also always non-NULL (because we allocated). I've added `ulValueLen != CK_UNAVAILABLE_INFORMATION` check instead.